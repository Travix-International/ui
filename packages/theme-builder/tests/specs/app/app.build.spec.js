import path from 'path';
import themeBuilder from '../../../src/app';

describe('App: build', () => {
  const defaultYaml = path.join(__dirname, '../mocks/default.yaml');
  const customYaml = path.join(__dirname, '../mocks/custom.yaml');
  const failYaml = path.join(__dirname, '../mocks/fail.yaml');

  it('should build yaml file to scss with default prefix', (done) => {
    const builder = themeBuilder({ format: 'scss' });

    builder.build(defaultYaml)
      .then((themeScss) => {
        expect(themeScss).toMatchSnapshot();
        done();
      });
  });

  it('should build yaml file to scss with "tx" prefix', (done) => {
    const builder = themeBuilder({ format: 'scss', prefix: 'tx' });

    builder.build(defaultYaml)
      .then((themeScss) => {
        expect(themeScss).toMatchSnapshot();
        done();
      });
  });

  it('should merge yaml files and build correct scss', (done) => {
    const builder = themeBuilder({ format: 'scss', prefix: 'tx' });

    builder.build([
      defaultYaml,
      customYaml,
    ])
      .then((themeScss) => {
        expect(themeScss).toMatchSnapshot();
        done();
      });
  });

  it('should throw error them yaml file is not valid', (done) => {
    const builder = themeBuilder({ format: 'scss', prefix: 'tx' });

    builder.build(failYaml)
      .catch((err) => {
        expect(err.message).toEqual('There are some errors in AST, please contact developer');
        done();
      });
  });
});
