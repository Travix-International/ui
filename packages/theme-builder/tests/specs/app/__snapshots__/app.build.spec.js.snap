// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App: build should build yaml file to scss with "tx" prefix 1`] = `
"$tx-generic-color-accent: #1fcff6;
$tx-button-color-bg: #1fcff6;
$tx-button-color-width: 125px;
$tx-button-border-style: rgba(31, 207, 246, 0.5);"
`;

exports[`App: build should build yaml file to scss with default prefix 1`] = `
"$generic-color-accent: #1fcff6;
$button-color-bg: #1fcff6;
$button-color-width: 125px;
$button-border-style: rgba(31, 207, 246, 0.5);"
`;

exports[`App: build should merge yaml files and build correct scss 1`] = `
"$tx-generic-color-accent: #283A8E;
$tx-button-color-bg: #283A8E;
$tx-button-color-width: 125px;
$tx-button-border-style: rgba(40, 58, 142, 0.5);"
`;
